core/async/src/tokio/runtime_handle.rs

```diff
@@ -0,0 +1,94 @@
+use std::sync::Arc;
+
+use tokio::sync::mpsc;
+
+use crate::futures::{DelayedActionRunner, FutureSpawner};
+use crate::messaging::Actor;
+
+/// TokioRuntimeMessage is a type alias for a boxed function that can be sent to the Tokio runtime.
+pub(super) struct TokioRuntimeMessage<A> {
+    pub(super) description: String,
+    pub(super) function: Box<dyn FnOnce(&mut A, &mut dyn DelayedActionRunner<A>) + Send>,
+}
+
+/// TokioRuntimeHandle is a handle to a Tokio runtime that can be used to send messages to an actor.
+/// It allows for sending messages and spawning futures into the Tokio runtime.
+pub struct TokioRuntimeHandle<A> {
+    /// The sender is used to send messages to the actor running in the Tokio runtime.
+    pub(super) sender: mpsc::UnboundedSender<TokioRuntimeMessage<A>>,
+    /// The runtime is the Tokio runtime that runs the actor and processes messages.
+    pub(super) runtime: Arc<tokio::runtime::Runtime>,
+}
+
+impl<A> Clone for TokioRuntimeHandle<A> {
+    fn clone(&self) -> Self {
+        Self { sender: self.sender.clone(), runtime: self.runtime.clone() }
+    }
+}
+
+impl<A> TokioRuntimeHandle<A>
+where
+    A: 'static,
+{
+    pub fn sender(&self) -> Arc<TokioRuntimeHandle<A>> {
+        Arc::new(self.clone())
+    }
+
+    pub fn future_spawner(&self) -> Box<dyn FutureSpawner> {
+        Box::new(self.clone())
+    }
+}
+
+/// Spawns an actor in the Tokio runtime and returns a handle to it.
+/// The handle can be used to get the sender and future spawner for the actor.
+///
+/// ```rust, ignore
+///
+/// struct MyActor;
+///
+/// impl Actor for MyActor {}
+///
+/// impl Handler<MyMessage> for MyActor {
+///     fn handle(&mut self, msg: MyMessage) {}
+/// }
+///
+/// // We can use the actor_handle to create senders and future spawners.
+/// let actor_handle = spawn_tokio_actor(MyActor);
+///
+/// let sender: MyAdapter = actor_handle.sender();
+/// let future_spawner = actor_handle.future_spawner();
+/// ```
+///
+/// The sender and future spawner can then be passed onto other components that need to send messages
+/// to the actor or spawn futures in the runtime of the actor.
+pub fn spawn_tokio_actor<A>(mut actor: A) -> TokioRuntimeHandle<A>
+where
+    A: Actor + Send + 'static,
+{
+    let runtime = tokio::runtime::Builder::new_multi_thread()
+        .worker_threads(1)
+        .enable_all()
+        .build()
+        .expect("Failed to create Tokio runtime");
+
+    let (sender, mut receiver) = mpsc::unbounded_channel::<TokioRuntimeMessage<A>>();
+
+    let runtime_handle = TokioRuntimeHandle { sender, runtime: Arc::new(runtime) };
+
+    // Spawn the actor in the runtime
+    let mut runtime_handle_clone = runtime_handle.clone();
+    runtime_handle.runtime.spawn(async move {
+        actor.start_actor(&mut runtime_handle_clone);
+        loop {
+            tokio::select! {
```

Comment:
This select makes me nervous. Could we just have message_opt = receiver.recv() and then match that message_opt? I'm not even sure what the behavior is the way it's currently written.

Also, how is "else" here even valid syntax; I've never seen this 😅

Reply:
Yeah, it's interesting, the `else` is invoked if one of the tokio select futures does return but the case doesn't match anything we have. Here, it would just amount to `receiver.recv()` matching `None`, but I've removed tokio select. We may need to reintroduce it when we add cancellation tokens but left it for now.